pulumi-deck/src/PulumiStageForm.tsx

```diff
@@ -0,0 +1,214 @@
+  }
+
+  public render() {
+    const { account, namespace } = this.props.formik.values['parameters'];
+    const { accounts, namespaces } = this.state;
+
+    return (
+      <>
+        <h4>Account and Namespace</h4>
+        <StageConfigField label="Account">
+          <AccountSelectInput
+            renderFilterableSelectThreshold={1}
+            value={account}
+            onChange={(evt: any) => this.handleAccountChange(evt.target.value)}
+            accounts={accounts}
+            provider="'kubernetes'"
+          />
+        </StageConfigField>
+        <StageConfigField label="Namespace">
+          <ReactSelectInput
+            clearable={false}
+            value={namespace}
+            options={namespaces.map((ns) => ({ value: ns, label: ns }))}
+            onChange={(evt: any) => this.handleNamespaceChange(evt.target.value)}
+          />
+        </StageConfigField>
+
+        <hr />
+
+        <h4>Repository Options</h4>
+        <FormikFormField
+          help={
+            <HelpField content="The working directory where the repository will be cloned into. All commands will be run in the context of this directory." />
+          }
+          input={(props) => <TextInput {...props} />}
+          label="Working Directory"
+          name="parameters.workingDir"
+        />
+        <FormikFormField
+          help={
+            <HelpField
+              content="The repo URL to clone from. For HTTPS URLs, specify the username and password as placeholders in the URL 
+            that map to the key names of secrets (see the Secrets Name parameter below). For SSH URLs, the plugin configuration must be
+            updated to specify the volume name for the SSH configuration files to use. The volume would be mounted as ~/.ssh in the
+            container."
+            />
+          }
+          input={(props) => <TextInput {...props} />}
+          label="Git Repo URL"
+          name="parameters.gitRepoUrl"
+        />
+        <FormikFormField
+          help={<HelpField content="The command to execute in order to restore dependencies for the Pulumi app." />}
+          input={(props) => <TextInput {...props} />}
+          label="Restore Dependencies Command"
+          name="parameters.restoreDepsCmd"
+        />
+
+        <hr />
+
+        <h4>Pulumi Options</h4>
+        <FormikFormField
+          help={
+            <HelpField
+              content="The container image to use for the RunJob. You may use a private image or one of Pulumi's public images.
+            The image must contain the Pulumi CLI as well as the language runtime used your Pulumi apps."
+            />
+          }
+          input={(props) => <TextInput {...props} />}
+          label="Container Image"
+          name="parameters.containerImage"
+        />
+        <FormikFormField
+          help={<HelpField content="The Pulumi stack name against which the command should be executed." />}
```

Comment:
This is confusing. I thought this was configuring if we're running up/refresh/etc.

Reply:
😬 messed up the help text here. It should be about the command. We could make the stack name a separate param so users don't have to remember to pass the `-s` flag in the args input. What do you think?

Reply:
That'd be helpful as long as it's not breaking and plays nice if I also specify `-s` in the commands. So I think we'd want the following scenarios to all be true:
- If I currently use the plugin and pass "-s stack" it should continue to work
- If I adopt the new stack parameter, it will work
- If I specify both -s via the command and via the stack parameter, the one passed via the stack parameter should take precedence (since it's newer, so the assumption is that if you use this, you meant it)

I'll approve this PR first and let's do a follow-up for the stack param if it's possible to do it in the way above.

Reply:
> I'll approve this PR first and let's do a follow-up for the stack param if it's possible to do it in the way above.

Sounds good!